src/pages/patientView/clinicalInformation/PatientViewPageStore.ts

```diff
@@ -435,6 +451,123 @@ export class PatientViewPageStore {
         []
     );
 
+    readonly resourceDefinitions = remoteData({
+        invoke: () =>
+            client.getAllResourceDefinitionsInStudyUsingGET({
+                studyId: this.studyId,
+            }),
+        onResult: defs => {
+            if (defs) {
+                for (const def of defs)
+                    if (def.openByDefault)
+                        this.setResourceTabOpen(def.resourceId, true);
+            }
+        },
+    });
+
+    readonly studyResourceData = remoteData<ResourceData[]>({
+        await: () => [this.resourceDefinitions],
+        invoke: () => {
+            const ret: ResourceData[] = [];
+            const studyResourceDefinitions = this.resourceDefinitions.result!.filter(
+                d => d.resourceType === 'STUDY'
+            );
+            const promises = [];
+            for (const resource of studyResourceDefinitions) {
+                promises.push(
+                    client
+                        .getAllStudyResourceDataInStudyUsingGET({
+                            studyId: this.studyId,
+                            resourceId: resource.resourceId,
+                            projection: 'DETAILED',
+                        })
+                        .then(data => ret.push(...data))
+                );
+            }
+            return Promise.all(promises).then(() => ret);
```

Comment:
I think promise.all should already return an array of the results of promises in the order called?   you could then just do 
`Promise.all(studyResourceDefinision.map(()=>clintCalls)`

Reply:
its a flat list in the end. could be `return _.flatten(await Promise.all(...))` but idk if that's easier to read